js/base-util.js

```diff
@@ -105,52 +100,179 @@ function baseUtil () {
+		if (d20plus.cfg.getOrDefault("chat", "resizeSidebarElements")) {
+			$("#rightsidebar").on("mouseout", d20plus.ut.resizeSidebar);
+			$("body").on("mouseup", d20plus.ut.resizeSidebar);
+			d20plus.ut.resizeSidebar("startup");
+		}
+		setTimeout(() => {
+			$(`.lamer-chat`).css("height", "0px");
+			setTimeout(() => {
+				$(".lamer-chat").remove();
+				clearInterval(d20plus.ut.cursor);
+			}, 2000);
+		}, 6000);
 	};
 
-	d20plus.ut.showLoadingMessage = (message) => {
-		const isStreamer = !!d20plus.cfg.get("interface", "streamerChatTag");
-		const scriptName = isStreamer ? "Script" : message;
-		d20plus.ut.sendHackerChat(`
-			${scriptName} initialising, please wait...<br><br>
-		`);
-		if (!window.enhancementSuiteEnabled) d20plus.ut.showHardDickMessage(scriptName);
+	d20plus.ut.showInitMessage = () => {
+		const consTemplate = `<div class="lamer-chat">
+			<span><span>&gt;</span>initializing, please wait...</span>
+			<span id="lamer-progress"><span>&gt;</span>loading data</span>
+			<span id="lamer-cursor"><span>&gt;</span><aside>|</aside></span>
+			<style type="text/css">
+			.lamer-chat {
+				font-family: Menlo, Monaco, Consolas, monospace;
+				font-size: small;
+				background: black;
+				display: inline-block;
+				font-weight: bold;
+				color: rgb(32, 194, 14);
+				width: 100%;
+				position: sticky;
+				z-index: 1000;
+				top: 0px;
+				height: 100px;
+				transition: height 2s;
+				overflow: hidden;
+				border-bottom: 1px solid rgb(32, 194, 14);
+			}
+			.lamer-chat > span {
+				display: block; white-space: nowrap;
+				padding: 0px 5px 0px 45px;
+			}
+			.lamer-chat > span > span {
+				float: left; margin-left: -39px;
+			}
+			</style>
+		</div>`;
+		$(`#textchat`).prepend(consTemplate);
+		let blink = false;
+		d20plus.ut.cursor = setInterval(() => {
+			$(`.lamer-chat`).append($(`.lamer-cursor`));
+			if (blink) $(`.lamer-chat aside`).html("|");
+			else $(`.lamer-chat aside`).html("");
+			blink = !blink;
+		}, 300);
 	};
 
+	d20plus.ut.showLoadingMessage = () => {
+		const isStreamer = !!d20plus.cfg?.get("chat", "streamerChatTag");
+		const scriptName = isStreamer ? "Script" : d20plus.scriptName;
+		const loadmsgtemplate = `<span><span>&gt;</span>loading ${d20plus.scriptName}</span>`;
+		if (!isStreamer) $(".lamer-chat > span:first-child").after(loadmsgtemplate);
+		if (!window.enhancementSuiteEnabled) d20plus.ut.showHardDickMessage(scriptName);
+	}
+
 	d20plus.ut.sendHackerChat = (message, error = false) => {
+		const legacytStyle = !!d20plus.cfg.get("chat", "legacySystemMessagesStyle");
+		if (!message) return;
 		d20.textchat.incoming(false, ({
 			who: "system",
-			type: "system",
-			content: `<span class="${error ? "hacker-chat-error" : "hacker-chat"}">
+			type: !legacytStyle && error ? "error" : "system",
+			content: (legacytStyle ? `<span class="${error ? "hacker-chat-error" : "hacker-chat"}">
 				${message}
-			</span>`,
+			</span>` : message),
 		}));
 	};
 
+	d20plus.ut.plantVersionInfo = () => {
```

Comment:
🤔 saving "yeah I'm running this dubious script" directly to the game's player "DB" seems like giving Roll20 a fast way to come up with a list of bannables

Reply:
we already have some clear indication we're running the script from the b20_ prefixed config vars (they were there before me). And the handouts also shouldn't be too hard to filter out if they wanted. I'm not insisting tho, this can be easily removed. I just thought it would be nice to have a way to tell who's running what for troubleshooting

Reply:
fair enough fair enough

Reply:
After some consideration I've rewritten the code so that gm may request versions via /v (name) command that will effectively send whisper, and other player's b20 will automatically whisper back. So script versions are no longer stored in the player's DB.